### PR TITLE
API-7095 - Publish test image

### PR DESCRIPTION
**Pull request by beaugrantham-va**
https://vajira.max.gov/browse/API-7095

Build and publish an `oauth-proxy-tests` image as part of CICD. This image will be hooked into the deploy process in a future story, allowing for fully automated regression tests.

### CODE REVIEW

**Review by dbrown-va — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -56,24 +55,24 @@ build/oauth:
...
+build/oauth_tests:
 	## build:	Build Docker image
 	docker build -t $(REPOSITORY)/$(NAMESPACE)/$(IMAGE):$(TAG) \
-		-f $(IMAGE)/DockerfileFG \
+		-f $(patsubst %-tests,%,$(IMAGE))/Dockerfile.bats \
 		--build-arg AWS_ACCOUNT_ID=$(AWS_ACCOUNT_ID) \
 		--build-arg BUILD_DATE_TIME=$(BUILD_DATE_TIME) \
 		--build-arg BUILD_TOOL=$(BUILD_TOOL) \
-		--build-arg VERSION=$(BUILD_VERSION) \
+		--build-arg BUILD_VERSION=$(BUILD_VERSION) \
```
Comment (beaugrantham-va) [inline Makefile:68]: Had to rename this from `VERSION` to something else. It was conflicting with the docker install script within the oauth-proxy-tests image. I made sure it still populates the dynamic image labels as expected.

**Review by duganth-va — APPROVED** [review]
Looks good to me! Only thing I might suggest is another dockerignore file if you want to isolate this container's build context.

**Comment by dbrown-va** [comment]
> Looks good to me! Only thing I might suggest is another dockerignore file if you want to isolate this container's build context.

Added dockerignore that excludes everything but the bats tests and the entrypoint. 
https://github.com/department-of-veterans-affairs/lighthouse-oauth-proxy/pull/12/commits/7e731c78c6407149523025a267e0ebd65ccb9c55

**Review by dbrown-va — APPROVED** [review]
Confirmed newest commit with dockerignore still build successfully and image is in ECR.

**Pull request merged**